drivers/espi/espi_mchp_xec.c

```diff
@@ -0,0 +1,822 @@
+		}
+	}
+}
+
+static void espi_xec_vw_isr(void *arg)
+{
+	struct device *dev = (struct device *)arg;
+	const struct espi_xec_config *config = dev->config->config_info;
+	u32_t girq_result;
+
+	girq_result = MCHP_GIRQ_RESULT(config->vw_girq_id);
+	REG32(MCHP_GIRQ_SRC_ADDR(config->vw_girq_id)) = girq_result;
+
+	for (int i = 0; i < m2s_vwires_isr_cnt; i++) {
+		struct espi_isr entry = m2s_vwires_isr[i];
+
+		if (girq_result & entry.girq_bit) {
+			if (entry.the_isr != NULL) {
+				entry.the_isr(dev);
+			}
+		}
+	}
+}
+
+static void espi_xec_periph_isr(void *arg)
+{
+	struct device *dev = (struct device *)arg;
+	const struct espi_xec_config *config = dev->config->config_info;
+	u32_t girq_result;
+
+	girq_result = MCHP_GIRQ_RESULT(config->pc_girq_id);
+	REG32(MCHP_GIRQ_SRC_ADDR(config->pc_girq_id)) = girq_result;
+
+	for (int i = 0; i < periph_isr_cnt; i++) {
+		struct espi_isr entry = peripherals_isr[i];
+
+		if (girq_result & entry.girq_bit) {
+			if (entry.the_isr != NULL) {
+				entry.the_isr(dev);
+			}
+		}
+	}
+}
+
+static int espi_xec_init(struct device *dev);
+
+static const struct espi_driver_api espi_xec_driver_api = {
+	.config = espi_xec_configure,
+	.get_channel_status = espi_xec_channel_ready,
+	.read_request = espi_xec_read_request,
+	.write_request = espi_xec_write_request,
+	.send_vwire = espi_xec_send_vwire,
+	.receive_vwire = espi_xec_receive_vwire,
+	.send_oob = espi_xec_send_oob,
+	.receive_oob = espi_xec_receive_oob,
+	.flash_read = espi_xec_flash_read,
+	.flash_write = espi_xec_flash_write,
+	.flash_erase = espi_xec_flash_erase,
+	.manage_callback = espi_xec_manage_callback,
+};
+
+static struct espi_xec_data espi_xec_data;
+static const struct espi_xec_config espi_xec_config = {
+	.base_addr = DT_MICROCHIP_XEC_ESPI_0_BASE_ADDRESS,
+	.bus_girq_id = MCHP_ESPI_IO_GIRQ,
+	.vw_girq_id = MEC_ESPI_MSVW_00_06_GIRQ,
+	.pc_girq_id = MCHP_KBC_GIRQ,
+};
+DEVICE_AND_API_INIT(espi_xec, DT_MICROCHIP_XEC_ESPI_0_LABEL, &espi_xec_init,
+			&espi_xec_data, &espi_xec_config,
+			PRE_KERNEL_2, CONFIG_ESPI_INIT_PRIORITY,
+			&espi_xec_driver_api);
+
+static int espi_xec_init(struct device *dev)
+{
+	const struct espi_xec_config *config = dev->config->config_info;
+	struct espi_xec_data *data =
+		(struct espi_xec_data *) (dev->driver_data);
+
+	data->plt_rst_asserted = 0;
+
+	/* Configure eSPI_PLTRST# to cause nSIO_RESET reset */
+	PCR_REGS->PWR_RST_CTRL = MCHP_PCR_PR_CTRL_USE_ESPI_PLTRST;
+	ESPI_CAP_REGS->PLTRST_SRC = MCHP_ESPI_PLTRST_SRC_IS_VW;
+
+	/* Configure the channels and its capabilities based on build config */
+	ESPI_CAP_REGS->GLB_CAP0 |= MCHP_ESPI_GBL_CAP0_VW_SUPP;
+	ESPI_CAP_REGS->GLB_CAP0 |= MCHP_ESPI_GBL_CAP0_PC_SUPP;
+
+	/* Max VW count is 12 pairs */
+	ESPI_CAP_REGS->VW_CAP = 11;
+	ESPI_CAP_REGS->PC_CAP |= MCHP_ESPI_PC_CAP_MAX_PLD_SZ_64;
+
+#ifdef CONFIG_ESPI_CHANNEL_OOB
+	ESPI_CAP_REGS->GLB_CAP0 |= MCHP_ESPI_GBL_CAP0_OOB_SUPP;
+	ESPI_CAP_REGS->OOB_CAP |= MCHP_ESPI_OOB_CAP_MAX_PLD_SZ_73;
+	ESPI_OOB_REGS->TX_ADDR_MSW = 0x1FFFul;
```

Comment:
I'm not sure if this should stay in hex or behind a mask. What would be the implication if this is changed to a Macro?

Reply:
better as a macro indeed, same for the numbers below